rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/TestBookKeeper.java

```diff
@@ -40,33 +38,29 @@
 {
   private static final Log log = LogFactory.getLog(TestBookKeeper.class);
 
-  private MetricRegistry metrics;
-  private Configuration conf;
+  private static final String TEST_CACHE_DIR_PREFIX = BookKeeperTestUtils.getTestCacheDirPrefix("bookKeeperTest");
+  private static final String TEST_DNE_CLUSTER_MANAGER = "com.qubole.rubix.core.DoesNotExistClusterManager";
+  private static final int TEST_MAX_DISKS = 1;
+
+  private final Configuration conf = new Configuration();
+  private final MetricRegistry metrics = new MetricRegistry();
 
   @BeforeMethod
   public void setUp() throws Exception
   {
-    conf = new Configuration();
-    metrics = new MetricRegistry();
-
     // Set configuration values for testing
-    CacheConfig.setCacheDataDirPrefix(conf, "/tmp/media/ephemeral");
-    CacheConfig.setMaxDisks(conf, 1);
-
-    // Create cache directories
-    Files.createDirectories(Paths.get(CacheConfig.getCacheDirPrefixList(conf)));
-    for (int i = 0; i < CacheConfig.getCacheMaxDisks(conf); i++) {
-      Files.createDirectories(Paths.get(CacheConfig.getCacheDirPrefixList(conf) + i));
-    }
+    CacheConfig.setCacheDataDirPrefix(conf, TEST_CACHE_DIR_PREFIX);
+
+    BookKeeperTestUtils.createCacheParentDirectories(CacheConfig.getCacheDirPrefixList(conf), TEST_MAX_DISKS);
   }
 
   @AfterMethod
   public void tearDown() throws Exception
   {
-    for (int i = 0; i < CacheConfig.getCacheMaxDisks(conf); i++) {
-      Files.walkFileTree(Paths.get(CacheConfig.getCacheDirPrefixList(conf) + i), new DeleteFileVisitor());
-      Files.deleteIfExists(Paths.get(CacheConfig.getCacheDirPrefixList(conf) + i));
-    }
+    BookKeeperTestUtils.removeCacheParentDirectories(CacheConfig.getCacheDirPrefixList(conf), TEST_MAX_DISKS);
```

Comment:
Does this go in AfterMethod or AfterClass ?

Reply:
@abhishekdas99 Moved to AfterClass for consistency, based on your other comments.